javascript-view-swapping/main.js

```diff
@@ -0,0 +1,27 @@
+var tabContainerElement = document.querySelector('.tab-container');
+var tabElements = document.querySelectorAll('.tab');
+var viewElements = document.querySelectorAll('.view');
+
+tabContainerElement.addEventListener('click', tabElementHandle);
+
+function tabElementHandle(event) {
+  if (event.target.matches('.tab')) {
+    // iterates through .tab NodeList and updates active class to swap tab styling
+    for (var i = 0; i < tabElements.length; i++) {
+      if (tabElements[i] === event.target) {
+        tabElements[i].className = 'tab active';
+      } else {
+        tabElements[i].className = 'tab';
+      }
+    }
+  }
+  // iterates through .view NodeList and updates hidden class to swap tabs
+  var attribute = event.target.getAttribute('data-view');
```

Comment:
This chunk of code should be inside of your conditional as well. As it is right now, this code will run whether the event target has the `tab` class or not 👍